headers/data/h_interpolator.hpp

```diff
@@ -62,64 +64,42 @@ class h_interpolator {
     h_interpolator();
     ~h_interpolator();
     double f( double );
+    double f_deriv( double );
     void newdata( int, double*, double* );
     void set_method( interpolation_methods );
 };
 
 inline void h_interpolator::locate(double x, int &iprev, int &inext) const
 {
-    if(ilast>=0) {
-        iprev = ilast;
-        // search near the last-used value
-        if(xdata[iprev] <= x) {
-            inext = iprev+1;
-            if(iprev == ndata-1 || xdata[inext] >= x)
-                return;         // found it in the same interval as last time, no need to update ilast
-            else
-                inext = ndata-1; // fall through to bisection search
-        }
-        else {
-            // check the previous interval
-            inext = iprev--;
-            if(inext==0)
-                return;         // don't set ilast to iprev; since iprev<0
-            else if(xdata[iprev] <= x) {
-                ilast = iprev;  // iprev >=0 in this case
-                return;
-            }
-            else
-                iprev = 0;      // fall through to search
-        }
-    }
-
```

Comment:
Was there something wrong with this code?  It was there to handle the case (common in ODE integrations) that the interval in a table interpolation moves slowly forward through the table, with occasional backward steps.  In practice it doesn't seem to be a particularly important optimization in Hector, so I'm not opposed to removing it; I'm just wondering what the motivation was.